rest/models/base.go

```diff
@@ -2,11 +2,13 @@ package models
 
 import (
   "time"
+  "gorm.io/gorm"
 )
 
 // Generic Struct used throughout models in this service.
 type BaseModel struct {
   Id uint
   CreatedAt time.Time
   UpdatedAt time.Time
+  DeletedAt gorm.DeletedAt
```

Comment:
Is it possible to use `time.Time` as well? The issue with the gorm type is that indexes the field as well and causes issue with OpenAPI within spec.

Reply:
For sure! if I understand this correctly, you want me to go from gorm.DeletedAt, to just a simple time stamp as well?

Reply:
Yeah. The gorm.DeletedAt is a bit problematic. We will see how it turns out.